packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/hierarchy/SubscriptionHierarchyEventDataFetcher.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.controllers.hierarchy;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.Objects;
+
+import org.eclipse.sirius.components.annotations.spring.graphql.SubscriptionDataFetcher;
+import org.eclipse.sirius.components.collaborative.charts.HierarchyConfiguration;
+import org.eclipse.sirius.components.collaborative.charts.HierarchyEventInput;
+import org.eclipse.sirius.components.collaborative.charts.IHierarchyEventProcessor;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.graphql.api.IDataFetcherWithFieldCoordinates;
+import org.eclipse.sirius.components.graphql.api.IEventProcessorSubscriptionProvider;
+import org.eclipse.sirius.components.graphql.api.IExceptionWrapper;
+import org.reactivestreams.Publisher;
+
+import graphql.schema.DataFetchingEnvironment;
+
+/**
+ * The data fetcher used to send the refreshed hierarchy to a subscription.
+ *
+ * @author pcdavid
+ */
+@SubscriptionDataFetcher(type = "Subscription", field = "hierarchyEvent")
+public class SubscriptionHierarchyEventDataFetcher implements IDataFetcherWithFieldCoordinates<Publisher<IPayload>> {
```

Comment:
Why is that in a test folder?

Reply:
We don't have a `sirius-components-charts-graphql` module yet and it felt overkill to create one just to host this single class only used in a test, especially as we do not have actual UI to create hierarchy representations.